src/pocketmine/scheduler/ClosureTask.php

```diff
@@ -40,20 +40,30 @@ class ClosureTask extends Task{
 
 	/** @var \Closure */
 	private $closure;
+	/** @var \Closure|null */
+	private $cancel;
 
 	/**
 	 * @param \Closure $closure Must accept only ONE parameter, $currentTick
+	 * @param \Closure|null $cancel Will be called if the first argument will return false
 	 */
-	public function __construct(\Closure $closure){
-		Utils::validateCallableSignature(function(int $currentTick) : void{}, $closure);
+	public function __construct(\Closure $closure, ?\Closure $cancel = null){
+		Utils::validateCallableSignature(($nullable = is_null($cancel)) ? function(int $currentTick) : void{} : function(int $currentTick) : bool{ return true; }, $closure);
+		if(!$nullable){
+			Utils::validateCallableSignature(function() : void{}, $cancel);
+		}
 		$this->closure = $closure;
+		$this->cancel = $cancel;
 	}
 
 	public function getName() : string{
 		return Utils::getNiceClosureName($this->closure);
 	}
 
 	public function onRun(int $currentTick){
-		($this->closure)($currentTick);
+		if(($this->closure)($currentTick) === false && $this->cancel !== null){
+			$this->getHandler()->cancel();
+			($this->cancel)();
```

Comment:
This can be called from inside `onCancel()` in line with standard tasks.

Reply:
Okey, I'll move it tommorow. That about is_null?

Reply:
Prefer comparison to `null` instead of that.